### ISSUE TITLE
Do not disturb is automatically set

### ISSUE THREAD

**Issue by lolokraus** [pachi81/GlucoDataHandler#188]
When you have the case that do not disturb is currently deactivated but the phone is in silent mode.
Then an alarm is triggered and you dismiss the alarm. In this case do not disturb is activated by the app even though it was disabled before.

**Comment by pachi81** [comment]
Ok, thank you! I will test.

**Comment by pachi81** [comment]
Not sure, what you mean, my test with force alarm sound:
No DnD, Silent -> Alarm disable Silent -> Silent is re-created, DnD stays disabled

DnD+Silent -> Alarm disable Silent+DnD -> Silent+DnD activated again

So what is your setup and what exact is the problem?

**Comment by lolokraus** [comment]
I am on Android 15, Pixel 8 and my steps to reproduce are exactly as you described in the first part, only difference is it enables DnD on my phone.

It seems to relate to this: https://stackoverflow.com/questions/58044974/enable-silent-mode-in-android-without-triggering-do-not-disturb

**Comment by pachi81** [comment]
Ok, I have to verify in emulator. Thank you for providing already a solution. At the moment, I´m testing new version 1.3, so I will add it.

**Comment by lolokraus** [comment]
Thank you! 
I have additional problems getting my phone to vibrate when the current setting is silent, I may look at it in the next days if I can identify the problem :)

**Comment by pachi81** [comment]
Great, vibration is something, I can not test in the emulator...

**Comment by pachi81** [comment]
@lolokraus Can you do another test for me, because in emulator it seems, that there is also a problem if the ringermode is only set to vibration:
- Enable force alarm sound
- Select an alarm without sound (or set sound to silent), but keep vibration active
- DnD off, Silent on
- Trigger test alarm
- Check ringer mode
In Emulator, it the result would be DnD on, Vibration on. With you fix, it would be DnD off, but Vibration on, instead of Silent.
I have a workaround to set ringermode to normal, before set to silent, but want to make sure, if this behavior is the same in real world...

**Comment by lolokraus** [comment]
@pachi81 I tested it and I had the same behavior you described in real life. I also tested the fix you already merged and it then worked correctly. Thx

**Comment by pachi81** [comment]
@lolokraus Thank you very much for testing!!! I wanted to provide you a test version, but you already did it! Great! Thank you very much!

**Comment by lolokraus** [comment]
No worries, and really great app you provide here. I will soon look a bit at the vibrations because they don't seem to work correctly on my phone, but if you need some help anytime you can write me.

**Comment by pannal** [comment]
Hmm please reopen this. I have the exact same issue with the latest version on Android 16 with my Moto Edge 60 Pro. Whenever there is an alarm from GDH (I have them set to fullscreen), DND is activated afterwards.

This is true for test alarms as wlel.

I've got this set:
- Fullscreen notification
- force alarm sound
- "use alarm clock volume"

**Comment by pannal** [comment]
OK, this setting is what causes it: "force alarm sound". When I turn that off, DND doesn't get engaged. It seems like current DND isn't detected properly, because in none of my tests did I have DND on before the test alarm.

**Comment by pachi81** [comment]
Force alarm sound disable DnD. But it is normally not needed, as DnD has exceptions for apps and alarm sounds. So I'm not using this setting..

**Comment by pannal** [comment]
True. I guess I don't need it, either. It's still a bug, though :)

**Comment by pachi81** [comment]
But what was is the problem, GDH only enables DnD, if it was active before!? If not, it is a bug.
Then I would need the logs to analyse. So please enable logging and debug-logging and reproduce it. So I can check.

**Comment by pannal** [comment]
It activates DND when it was NOT on :)

**Comment by pachi81** [comment]
So please try to reproduce with logging. 
Thank you.

**Comment by pannal** [comment]
[GDH_PHONE_APP_20260709_225835.txt](https://github.com/user-attachments/files/29865944/GDH_PHONE_APP_20260709_225835.txt)

**Comment by pannal** [comment]
That's a test notification with debug logging on, no DND beforehands, but DND afterwards.

**Comment by pachi81** [comment]
Thank you! Have to check.

**Comment by pachi81** [comment]
Ok, it seems to be a bug(?) or special implementation of Motorola. 
GDH requests the current ringer mode and the phone says "RINGER_MODE_SILENT", so it looks like DnD enabled. 
If I do not handle it as DnD I could not force sound, so this is more a problem.
But good to know, maybe I should leave an hint, that it is better to user the Android DnD settings instead of "Force" settings in GDH.
